src/Microsoft.CodeQuality.Analyzers/UnitTests/Maintainability/AvoidUninstantiatedInternalClassesTests.cs

```diff
@@ -1346,6 +1346,154 @@ internal class CFoo {}
+    public class MyTextBox { }
+}");
+        }
+
+        [Fact, WorkItem(2957, "https://github.com/dotnet/roslyn-analyzers/issues/2957")]
+        public async Task CA1812_DesignerAttributeGlobalTypeName_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+using System.ComponentModel;
+
+internal class MyTextBoxDesigner { }
+
+[Designer(""MyTextBoxDesigner, TestProject"")]
+public class MyTextBox { }");
+        }
+
+        [Fact, WorkItem(2957, "https://github.com/dotnet/roslyn-analyzers/issues/2957")]
+        public async Task CA1812_DesignerAttributeType_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+using System.ComponentModel;
+
+namespace Foo
+{
+    internal class MyTextBoxDesigner { }
+
+    [Designer(typeof(MyTextBoxDesigner))]
+    public class MyTextBox { }
+}");
+        }
+
+        [Fact, WorkItem(2957, "https://github.com/dotnet/roslyn-analyzers/issues/2957")]
+        public async Task CA1812_DesignerAttributeTypeNameWithBaseTypeName_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+using System.ComponentModel;
+
+namespace Foo
+{
+    public class SomeBaseType { }
+    internal class MyTextBoxDesigner { }
+
+    [Designer(""Foo.MyTextBoxDesigner, TestProject"", ""Foo.SomeBaseType"")]
+    public class MyTextBox { }
+}");
+        }
+
+        [Fact, WorkItem(2957, "https://github.com/dotnet/roslyn-analyzers/issues/2957")]
+        public async Task CA1812_DesignerAttributeTypeNameWithBaseType_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+using System.ComponentModel;
+
+namespace Foo
+{
+    public class SomeBaseType { }
+    internal class MyTextBoxDesigner { }
+
+    [Designer(""Foo.MyTextBoxDesigner, TestProject"", typeof(SomeBaseType))]
+    public class MyTextBox { }
+}");
+        }
+
+        [Fact, WorkItem(2957, "https://github.com/dotnet/roslyn-analyzers/issues/2957")]
+        public async Task CA1812_DesignerAttributeTypeWithBaseType_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+using System.ComponentModel;
+
+namespace Foo
+{
+    public class SomeBaseType { }
+    internal class MyTextBoxDesigner { }
+
+    [Designer(typeof(Foo.MyTextBoxDesigner), typeof(SomeBaseType))]
+    public class MyTextBox { }
+}");
+        }
+
+        [Fact, WorkItem(2957, "https://github.com/dotnet/roslyn-analyzers/issues/2957")]
+        public async Task CA1812_DesignerAttributeNestedTypeName_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+using System;
+using System.ComponentModel;
+
+namespace Foo
+{
+    [Designer(""Foo.MyTextBox.MyTextBoxDesigner, TestProject"")]
+    public class MyTextBox
+    {
+        internal class MyTextBoxDesigner { }
+    }
+}",
+                // False-Positive: when evaluating the string of the DesignerAttribute the type symbol doesn't exist yet
```

Comment:
@mavasani FYI we no longer properly support the case of a nested-type referenced by string from an outer designer attribute.

Reply:
I think this is fine.